Add test stub for common tests and DataOperationTests

dadlex opened the pull request:
Add stub where we create a pravega standalone container and then we run some empty tests. After creation of the container we call a constructor
of pravega storage driver.
After that the pravega docker container is killed.

Besides, the map check in invokePathRead was deleted as useless. StreamManager was deleted from class fields, as it is unavailable to use it this way (on Pravega level).

dadlex committed:
Add test stub for common tests and DataOperationTests

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Add stub where we create a pravega standalone container and then we run some empty tests. After creation of the container we call a constructor
of pravega storage driver.
After that the pravega docker container is killed.

Signed-off-by: Alexander <alex15051997@yandex.com>

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -280,8 +273,9 @@ private boolean invokePathRead(final PathOperation<? extends PathItem> pathOp) {
 
 	protected boolean invokePathDelete(final PathOperation<? extends PathItem> pathOp) {
 		final String path = pathOp.dstPath();
+		StreamManager streamManager = StreamManager.create(URI.create(uriSchema));
```
it may be useful to let it be final

dadlex replied:
Agree

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/integration/DataOperationsTest.java:
```diff
@@ -0,0 +1,201 @@
+			throws Exception {
+
+		}
+}
```
GH diff warns about a missing newline here

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/util/docker/PravegaNodeContainer.java:
```diff
@@ -0,0 +1,60 @@
+			CONTAINER_ID = null;
+		}
+	}
+	}
```
GH diff warns about a missing newline here

dadlex replied:
My indents were a mess for some reason. I'll correct it in the next commit.

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/integration/CommonTest.java:
```diff
@@ -0,0 +1,158 @@
+		@AfterClass
+		public static void tearDownClass()
+			throws Exception {
+			PRAVEGA_NODE_CONTAINER.close();
```
```
./gradlew test
...
com.emc.mongoose.storage.driver.pravega.integration.CommonTest > classMethod FAILED
    com.github.dockerjava.api.exception.ConflictException at CommonTest.java:148
```

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/integration/DataOperationsTest.java:
```diff
@@ -0,0 +1,201 @@
+	public static void setUpClass()
+			throws Exception {
+		try {
+			PRAVEGA_NODE_CONTAINER = new PravegaNodeContainer();
```
```
./gradlew test
...
com.emc.mongoose.storage.driver.pravega.integration.DataOperationsTest > classMethod FAILED
    java.lang.AssertionError at DataOperationsTest.java:140
        Caused by: com.github.dockerjava.api.exception.BadRequestException at DataOperationsTest.java:138
```

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/integration/DataOperationsTest.java:
```diff
@@ -0,0 +1,201 @@
+	@AfterClass
+	public static void tearDownClass()
+			throws Exception {
+		PRAVEGA_NODE_CONTAINER.close();
```
```
./gradlew test
...
com.emc.mongoose.storage.driver.pravega.integration.DataOperationsTest > classMethod FAILED
    java.lang.NullPointerException at DataOperationsTest.java:147
```

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -57,13 +57,13 @@ public PravegaStorageDriver(
 		nodePort = storageConfig.intVal("net-node-port");
 		final List<String> endpointAddrList = nodeConfig.listVal("addrs");
-		readerTimeoutMs = storageConfig.intVal("storage-item-input-readerTimeout");
+		readerTimeoutMs = storageConfig.intVal("item-input-readerTimeout");
```
What do you think about to use `storage-read-timeoutMillis` instead `storage-item-input-readerTimeout`?

dadlex replied:
Do you suggest changing the path in JSON file or do you want to say I should use the "storage-net-timeoutMillis" JSON parameter? If it's the second one, then sure, I'll change it, I just wasn't sure if the "net-timeoutMillis" had something to do with the reader. If it's the first one, I'll correct it as well, sounds good to me.

akurilov replied:
I mean a new configuration option `storage-read-timeoutMillis` describing the Pravega-specific & read-specific timeout

evtimo committed:
Update travis

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

dadlex committed:
Add test stub for common tests and DataOperationTests

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Correct indents.
Make local streamManager in invokePathDelete() final.

Signed-off-by: Alexander <alex15051997@yandex.com>
